projects/imbu/README.md

```diff
@@ -0,0 +1,191 @@
+   image from the `Dockerfile` in `projects/imbu`, specifying the root of
+   `nupic.research` as the build context.  This is required for inclusion of
+   `htmresearch` in addition to `projects/imbu/`.
+- The repository is called `imbu`, and the image is tagged `latest`.
+- `docker run` starts a container from the image tagged `latest` in the `imbu`
+  repository.
+- `--name imbu` names the container `imbu`
+- `-d` causes the container to run in `daemon` mode in the background
+- `-p 8080:80` maps port 80 in the container to port 8080 on the host.
+- `-e IMBU_LOAD_PATH_PREFIX=${IMBU_LOAD_PATH_PREFIX}` specifies the location
+  from which pre-trained models will be loaded.
+- `-e CORTICAL_API_KEY=${CORTICAL_API_KEY}` defines the `CORTICAL_API_KEY`
+  environment variable in the container.  For the command above to work as-is
+  you must have `CORTICAL_API_KEY` set in your own environment!  You may specify
+  the value inline.
+- `-e IMBU_RETINA_ID=${IMBU_RETINA_ID}` defines the `IMBU_RETINA_ID`
+  environment variable in the container.  For the command above to work as-is
+  you must have `IMBU_RETINA_ID` set in your own environment!  You may specify
+  the value inline.
+
+Additionally, there are some optional `docker run` arguments:
+
+- Map the `/opt/numenta/imbu/cache` directory in the container to one on on the
+  host external to the container to persist model state across containers.  For
+  example:
+
+  ```
+  -v `pwd`/cache:/opt/numenta/imbu/cache
+  ```
+- You may use a local clone of nupic.research in the container by mapping the
+  `/opt/numenta/nupic.research` directory.  This will allow you to test changes
+  to nupic.research locally before commiting changes and pushing to github.
+
+  ```
+  -v <path to nupic.research>:/opt/numenta/nupic.research
+  ```
+
+A few helper utilities are included for your convenience:
+
+- `destructive-build-and-refresh-container.sh` to clear persistent model cache,
+  rebuild, and restart the container from scratch
+- `graceful-build-and-refresh-container.sh` to rebuild and restart the
+  container, preserving persistent model cache
+
+You may also set additional options to the `docker` command by specifying the
+`IMBU_DOCKER_OPTIONS` environment variable. Use `docker help` for a list of
+available options.
+
+Both of these helper utilities will pass along additional arguments to the
+`docker run` command specified in the `IMBU_DOCKER_EXTRAS` environment
+variable.  For example, to map the directories described above.  It can be
+helpful to define all relevant environment variables in a file that you can
+either `source` at-will, or in your bash profile.  For example:
+
+```
+export IMBU_DOCKER_EXTRAS="-v <path to cache>:/opt/numenta/imbu/cache -v <path to nupic.research>:/opt/numenta/nupic.research"
+export CORTICAL_API_KEY="<cortical.io api key>"
+export IMBU_RETINA_ID="en_associative"
+export IMBU_LOAD_PATH_PREFIX="/opt/numenta/imbu/engine"
```

Comment:
Based on https://github.com/numenta/nupic.research/pull/482#issuecomment-195107762 this variable should now be `/opt/numenta/nupic.research/projects/imbu/engine`